### PR TITLE
python3Packages.notobuilder: init at 0-unstable-2024-08-03

### PR DESCRIPTION
**Pull request by jopejoe1**
## Description of changes
Python modules required to build noto-fonts from source split out from #331341 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/ufomerge/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    ufolib2
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    fontfeatures
+  ];
+
+  disabledTestPaths = [
+    # Failing due to fonttools bein to old
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ufomerge/default.nix:46]: *being

File: pkgs/development/python-modules/opentype-feature-freezer/default.nix [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  src = fetchFromGitHub {
+    owner = "twardoch";
+    repo = "fonttools-opentype-feature-freezer";
+    rev = "2ae16853bc724c3e377726f81d9fc661d3445827";
+    hash = "sha256-mIWQF9LTVKxIkwHLCTVK1cOuiaduJyX8pyBZ/0RKIVE=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace poetry.masonry.api poetry.core.masonry.api \
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/opentype-feature-freezer/default.nix:27]: Use --replace-fail

File: pkgs/development/python-modules/paintcompiler/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    repo = "paintcompiler";
+    rev = "v${version}";
+    hash = "sha256-dmVBQUUyFc71zq8fXBQ4ii/llrGdtUCOponCzSeut6g=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    fonttools
+    black
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/paintcompiler/default.nix:26]: Why do we need a formatter?

Comment (jopejoe1) [reply pkgs/development/python-modules/paintcompiler/default.nix:26]: Because it can decompile COLR tables to python files and format these with black.

**Comment by a-kenji** [comment]
Result of `nixpkgs-review pr 333709` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>46 packages built:</summary>
  <ul>
    <li>bump2version (python312Packages.bump2version)</li>
    <li>bump2version.dist (python312Packages.bump2version.dist)</li>
    <li>nanoemoji (python312Packages.nanoemoji)</li>
    <li>nanoemoji.dist (python312Packages.nanoemoji.dist)</li>
    <li>openmoji-black</li>
    <li>openmoji-color</li>
    <li>python311Packages.blackrenderer</li>
    <li>python311Packages.blackrenderer.dist</li>
    <li>python311Packages.bump2version</li>
    <li>python311Packages.bump2version.dist</li>
    <li>python311Packages.bumpfontversion</li>
    <li>python311Packages.bumpfontversion.dist</li>
    <li>python311Packages.diffenator2</li>
    <li>python311Packages.diffenator2.dist</li>
    <li>python311Packages.gftools</li>
    <li>python311Packages.gftools.dist</li>
    <li>python311Packages.nanoemoji</li>
    <li>python311Packages.nanoemoji.dist</li>
    <li>python311Packages.notobuilder</li>
    <li>python311Packages.notobuilder.dist</li>
    <li>python311Packages.opentype-feature-freezer</li>
    <li>python311Packages.opentype-feature-freezer.dist</li>
    <li>python311Packages.paintcompiler</li>
    <li>python311Packages.paintcompiler.dist</li>
    <li>python311Packages.ufomerge</li>
    <li>python311Packages.ufomerge.dist</li>
    <li>python311Packages.vttlib</li>
    <li>python311Packages.vttlib.dist</li>
    <li>python312Packages.blackrenderer</li>
    <li>python312Packages.blackrenderer.dist</li>
    <li>python312Packages.bumpfontversion</li>
    <li>python312Packages.bumpfontversion.dist</li>
    <li>python312Packages.diffenator2</li>
    <li>python312Packages.diffenator2.dist</li>
    <li>python312Packages.gftools</li>
    <li>python312Packages.gftools.dist</li>
    <li>python312Packages.notobuilder</li>
    <li>python312Packages.notobuilder.dist</li>
    <li>python312Packages.opentype-feature-freezer</li>
    <li>python312Packages.opentype-feature-freezer.dist</li>
    <li>python312Packages.paintcompiler</li>
    <li>python312Packages.paintcompiler.dist</li>
    <li>python312Packages.ufomerge</li>
    <li>python312Packages.ufomerge.dist</li>
    <li>python312Packages.vttlib</li>
    <li>python312Packages.vttlib.dist</li>
  </ul>
</details>

**Pull request merged**